numba/targets/base.py

```diff
@@ -437,11 +437,40 @@ def get_attribute(self, val, typ, attr):
             offset = typ.offset(attr)
             elemty = typ.typeof(attr)
 
-            @impl_attribute(typ, attr, elemty)
-            def imp(context, builder, typ, val):
-                dptr = cgutils.get_record_member(builder, val, offset,
-                                                 self.get_data_type(elemty))
-                return self.unpack_value(builder, elemty, dptr)
+            if isinstance(elemty, types.NestedArray):
+                # Inside a structured type only the array data is stored, so we
+                # create an array structure to point to that data.
+                aryty = arrayobj.make_array(elemty)
+                @impl_attribute(typ, attr, elemty)
+                def imp(context, builder, typ, val):
+                    ary = aryty(context, builder)
+                    dtype = elemty.dtype
+                    ary.nitems = context.get_constant(types.intp, elemty.nitems)
+                    assert dtype.bitwidth % 8 == 0, ("Dtype bitwidth must be"
+                                                     "a multiple of bytes")
+                    size = dtype.bitwidth//8
+                    ary.itemsize = context.get_constant(types.intp, size)
+                    dataty = self.get_data_type(dtype)
+                    ary.data = cgutils.get_record_member(builder, val, offset,
+                                                         dataty)
+                    ary.shape = cgutils.pack_array(builder,
+                                                   [ self.get_constant(types.intp, s)
+                                                     for s in elemty.shape ])
+                    stride = size
+                    strides = []
```

Comment:
OK - moved them over to `NestedArray`. I don't think the size will vary depending on the target type for the present use case.